docs/en/cookbook/working-with-datetime.rst

```diff
@@ -177,7 +113,12 @@ requiring timezoned datetimes:
         public function getCreated()
         {
             if (!$this->localized) {
-                $this->created->setTimeZone(new \DateTimeZone($this->timezone));
+                
+                $class = $this->created::class;
+                $this->created = new $class(
```

Comment:
I suggest using one of either `DateTimeImmutable` or `DateTime`. Since DBAL 2.6, `DateTimeImmutable` is viable, so I suggest just going with that rather than with a generic example